Hero.h

```diff
@@ -8,19 +20,29 @@
 class Hero
 {
 private:
-	int hp;
-	const int dmg;
-	const std::string name;
+	int hp; ///< The starting HP of the heroes
+	const int dmg; ///< The damage of the heroes
+	const float atkspeed; ///< How many seconds pass between the attacks of the hero
+	const std::string name; ///< The name of the hero
+	float nextAttack=atkspeed; ///< Keeps track of the current attackspeed
+	void getAttack (const Hero& enemy); ///< A hero take damage from another hero
 
 public:
-	Hero(const std::string name, int hp, const int dmg) : name(name), hp(hp), dmg(dmg) {}
-
-	friend std::ostream& operator << (std::ostream& os, const Hero& hero);
-	void getAttack (const Hero& h);
-	std::string getName() const { return name; }
-	int getHp() const { return hp; }
-	int getDmg() const { return dmg; }
-	static Hero parseUnit (const std::string& filename);
+/**
+ * Setting the hero properties and the required things to the battle.
+*/
+	Hero(const std::string name, int hp, const int dmg, const float atkspeed) : name(name), hp(hp), dmg(dmg), atkspeed(atkspeed){} ///< Setting the heroes properties
+	friend std::ostream& operator << (std::ostream& os, const Hero& hero); /// operator overloading
+	std::string getName() const { return name; } ///< getter for the name of the hero
+	int getHp() const { return hp; } ///< getter for the hp of the hero
+	int getDmg() const { return dmg; } ///< getter for the damage of the hero
+	static Hero parseUnit (const std::string& filename); ///< reading the properties from file
+	float getAtkspeed() const { return atkspeed; } ///< getter for the attack speed of the hero
+	void setNextAttack() { nextAttack+=atkspeed; } ///< after an attack setting the nextattack as a "timer" 
+	float getNextAttack() const {return nextAttack; } ///< getter for the next attack of the hero
+	void fight(Hero& attacked); ///< the hero who get attacked
+	void takeAttack(Hero& enemy) { enemy.getAttack(*this); } ///< attacking the enemy

```

Comment:
A takeAtack függvény teljesen felesleges kódismétlés. Ha szeretnéd, felőlem maradhat, de anélkül is meg lehet oldani.